account_invoice_refund_reason/views/account_move_view.xml

```diff
@@ -0,0 +1,31 @@
+<odoo>
+
+    <record id="extended_view_account_invoice" model="ir.ui.view">
+        <field name="name">extended.account.invoice.form</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_move_form" />
+        <field name="arch" type="xml">
+            <xpath
+                expr="//page[@name='other_info']/group/group[2]/field[@name='fiscal_position_id']"
+                position="after"
+            >
+                <field
+                    name="reason_id"
+                    attrs="{'invisible': [('reason_id', '=', False)]}"
+                    readonly="1"
+                />
+            </xpath>
+            <xpath
+                expr="//page[@id='other_tab_entry']/group/group[1]/field[@name='to_check']"
+                position="after"
+            >
+                <field
+                    name="reason_id"
+                    attrs="{'invisible': [('reason_id', '=', False)]}"
+                    readonly="1"
+                />
+            </xpath>
```

Comment:
```suggestion
            <group name="sale_info_group" position="inside">
                <field name="reason_id" attrs="{'invisible': [('reason_id', '=', False)]}" />
            </group>
            <group name="accounting_info_group" position="inside">
                <field name="reason_id" attrs="{'invisible': [('reason_id', '=', False)]}" />
            </group>
```

To keep it simple :)
Note that I removed `readonly="1"` too.  It's better to make it readonly=True in the python model instead